R/viz.R

```diff
@@ -2354,18 +2356,23 @@ prettyForestPlot = function(maf,
       dplyr::rename("Tumor_Sample_Barcode"="sample_id")
   }else if(!missing(maf)){
     if(separate_hotspots){
-        if(!"hot_spot" %in% colnames(maf))
-          stop("No \"hot_spot\" column in maf file. Annotate your maf file with GAMBLR::annotate_hot_spots() first. ")
+      if(!"hot_spot" %in% colnames(maf))
+        stop("No \"hot_spot\" column in maf file. Annotate your maf file with GAMBLR::annotate_hot_spots() first. ")
       maf$Hugo_Symbol = ifelse(!is.na(maf$hot_spot), paste0(maf$Hugo_Symbol, "_hotspot"), maf$Hugo_Symbol)
     }
     #Convert the maf file to a binary matrix
     mutmat = maf %>%
       dplyr::select(Hugo_Symbol, Tumor_Sample_Barcode) %>%
-      left_join(dplyr::select(metadata, Tumor_Sample_Barcode, comparison), by = "Tumor_Sample_Barcode") %>%
+      full_join(dplyr::select(metadata, Tumor_Sample_Barcode, comparison), by = "Tumor_Sample_Barcode") %>%
       distinct() %>%
       dplyr::mutate(is_mutated = 1) %>%
       pivot_wider(names_from = Hugo_Symbol, values_from = is_mutated, values_fill = 0) %>%
       dplyr::mutate(across(where(is.numeric), ~replace_na(., 0)))
+    if(rm_na_samples && "NA" %in% colnames(mutmat)){
+      mutmat = mutmat %>%
+        dplyr::filter(`NA` == 0) #filter out all samples that show no mutations in the selected genes (i.e na_samples = 1).
+    }
+    mutmat = dplyr::select(mutmat, -`NA`)
```

Comment:
Sorry for missing this when we were chatting about this yesterday, but I think this still needs to be conditional because if the column `NA` isn't in the data frame this will error out. 
```
if("NA" %in% colnames(mutmat)){
    mutmat = dplyr::select(mutmat, -`NA`)
}
```